ClangSharp.PInvokeGenerator/OutputBuilderFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ClangSharp
+{
+    public sealed class OutputBuilderFactory
```

Comment:
I thought about making this, `OutputBuilder`, and `PInvokeGenerator` abstract and then explicitly having a `CSharpOutputBuilderFactory`, `CSharpOutputBuilder`, and `CSharpPInvokeGenerator`. This would allow people to more trivially support other languages (like F#), if they so desired and would work fairly well since most of the logic is around the traversal, not the text generation...